src/contexts/playlist-context.js

```diff
@@ -0,0 +1,127 @@
+import { createContext, useContext, useEffect, useState } from "react";
+import { useAxios } from "../utils";
+import { useAuth } from "./auth-context";
+
+const PlaylistContext = createContext();
+
+const usePlaylist = () => useContext(PlaylistContext);
+
+const PlaylistProvider = ({ children }) => {
+  const [playlistToShow, setPlaylistToShow] = useState([]);
+  const {
+    response: responsePlaylist,
+    loading: loadingPlaylist,
+    operation: operationPlaylist,
+  } = useAxios();
+
+  const fetchAllPlaylists = () => {
+    operationPlaylist({
+      method: "get",
+      url: "/api/user/playlists",
+      headers: {
+        accept: "*/*",
+        authorization: localStorage.getItem("cobraToken"),
+      },
+      data: {},
+    });
+  };
+
+  const { isUserLoggedIn } = useAuth();
+  useEffect(() => {
+    isUserLoggedIn ? fetchAllPlaylists() : setPlaylistToShow([]);
+  }, [isUserLoggedIn]);
+
+  const addPlaylist = (title, description = "") => {
+    operationPlaylist({
+      method: "POST",
+      url: "/api/user/playlists",
+      headers: {
+        accept: "*/*",
+        authorization: localStorage.getItem("cobraToken"),
```

Comment:
suggestion: instead of fetching `localStorage.getItem("cobraToken")` everytime, get it out once in a variable and use it wherever needed.

Reply:
Good suggestion! 🚀
Will take care of this while refactoring.